cypress/support/commands.js

```diff
@@ -23,3 +23,235 @@
+/**
+ * Add saved Dashboard panels
+ * @param {String} keyword Search term for panels of interest
+ * @param {String} type Panel type to search for
+ * @param {Boolean} multiplePages Whether there are multiple pages to iterate through
+ */
+Cypress.Commands.add('addDashboardPanels', (keyword, type, multiplePages = true) => {
+  const iteratePages = () => {
+    cy.get('[data-test-subj="pagination-button-next"]').then(($nextBtn) => {
+      cy.get('[data-test-subj="savedObjectFinderItemList"]').should('be.visible')
+      if ($nextBtn.is(':enabled')) {
+        cy.wrap($nextBtn).click()
+        cy.get('[data-test-subj^="savedObjectTitle' + keyword.replace(/\s+/g, '-') + '"]').should('be.visible').each(($button) => {
+          cy.wrap($button).click()
+        })
+        iteratePages()
+      }
+    })
+  }
+  cy.get('[data-test-subj="dashboardAddPanelButton"]').should('be.visible').click()
+  cy.get('[data-test-subj="savedObjectFinderItemList"]').should('be.visible')
+
+  cy.get('[data-test-subj="savedObjectFinderFilterButton"]').should('be.visible').click()
+  cy.get('[data-test-subj="savedObjectFinderFilter-' + type + '"]').should('be.visible').click()
+  cy.get('[data-test-subj="savedObjectFinderFilterButton"]').should('be.visible').click()
+
+  cy.get('[data-test-subj="savedObjectFinderSearchInput"]').should('be.visible').type(keyword)
+
+  cy.get('[data-test-subj^="savedObjectTitle' + keyword.replace(/\s+/g, '-') + '"]').should('be.visible').each(($button) => {
+    cy.wrap($button).click()
+  }).then(() => {
+    if (multiplePages) {
+      cy.wrap(iteratePages())
+    }
+  })
+  cy.get('[data-test-subj="euiFlyoutCloseButton"]').click()
+})
+
+/**
+ * Attempts to add a specified filter to a dashboard, retrying by reopening the filter window
+ * @param {String} field Field value to select
+ * @param {String} operator Operator value to select
+ * @param {String} value Value field input
+ */
+
+Cypress.Commands.add('addDashboardFilterRetryFull', (field, operator, value = null) => {
+  // Try and select the desire combo box option
+  const selectComboBoxInput = (selector, keyword) => {
+    cy.get('[data-test-subj="' + selector + '"]').find('[data-test-subj="comboBoxInput"]').trigger('focus').type('{selectall}{backspace}' + keyword)
+    cy.get('[data-test-subj="comboBoxOptionsList ' + selector + '-optionsList"]').find('[title="' + keyword + '"]').trigger('click', { force: true })
+  }
+
+  // Attempt up to three times to select the desired field and operator options and input the value (if applicable) 
+  const tryToAddFilter = (field, operator, value = null, retry = 0) => {
+    cy.wait(3000)
+    cy.get('[data-test-subj="addFilter"]').click({ scrollBehavior: 'center' }).then(() => {
+      
+      selectComboBoxInput('filterFieldSuggestionList', field)
+      cy.get('[data-test-subj="filterFieldSuggestionList"]').then(($field) => {
+        const cls = $field.attr('class')
+        if (cls.includes('euiComboBox-isInvalid') && retry < 3) {
```

Comment:
Nonblocking: better to create a constant for number 3 so that it won't be the "magic" number.

Reply:
I can implement this in the next PR. I also want to try and make constants for other values in future releases (such as some of the selector strings) to improve readability.